numpy/_core/fromnumeric.py

```diff
@@ -296,7 +304,27 @@ def reshape(a, newshape, order='C'):
            [3, 4],
            [5, 6]])
     """
-    return _wrapfunc(a, 'reshape', newshape, order=order)
+    if newshape is None and shape is None:
+        raise TypeError(
+            "reshape() missing 1 required positional argument: 'shape'")
```

Comment:
Interestingly, `shape=None` was previously supported as "do nothing", I don't know how that would be sensible, so probably doesn't matter.

Reply:
Ah, I don't think `shape=None` is useful really (I think some people could even expect it flattens rather than "do nothing"). I would stick to throwing an error on `shape=None`.